src/txn/txn_ckpt.c

```diff
@@ -2273,6 +2274,18 @@ __wt_checkpoint_tree_reconcile_update(WT_SESSION_IMPL *session, WT_TIME_AGGREGAT
             ckpt->run_write_gen = btree->run_write_gen;
             WT_TIME_AGGREGATE_COPY(&ckpt->ta, ta);
         }
+
+    /*
+     * Ensure the maximum timestamp used for reconciliation tracks the greatest value between the
+     * start and stop durable timestamps. The value may be higher than these two if eviction occurs
+     * in parallel. However, during RTS and recovery, we should never have it set higher or the
+     * system might think there is newer content that needs to be checkpointed.
+     */
+    max_start_stop_durable_ts = WT_MAX(ta->newest_start_durable_ts, ta->newest_stop_durable_ts);
+    if (F_ISSET(S2C(session), WT_CONN_RECOVERING) ||
+      F_ISSET(session, WT_SESSION_ROLLBACK_TO_STABLE) ||
+      btree->rec_max_timestamp < max_start_stop_durable_ts)
```

Comment:
I feel there is no need to compare `btree->rec_max_timestamp < max_start_stop_durable_ts`. We can directly reset the `btree->rec_max_timestamp` with `max_start_stop_durable_ts` when during recovery or RTS scenarios. 

In normal cases, the update to the `btree->rec_max_timestamp` is already taken care.